osu.Game/Screens/Edit/Components/Timelines/Summary/Parts/TimelinePart.cs

```diff
@@ -25,16 +25,36 @@ protected TimelinePart()
 
             Beatmap.ValueChanged += b =>
             {
-                timeline.Clear();
-                timeline.RelativeChildSize = new Vector2((float)Math.Max(1, b.Track.Length), 1);
+                updateRelativeChildSize();
                 LoadBeatmap(b);
             };
         }
 
+        private void updateRelativeChildSize()
+        {
+            if (!Beatmap.Value.TrackLoaded)
+            {
+                timeline.RelativeChildSize = Vector2.One;
+                return;
+            }
+
+            var track = Beatmap.Value.Track;
+
+            if (!track.IsLoaded)
+            {
+                // the track may not be loaded completely (only has a length once it is).
+                Schedule(updateRelativeChildSize);
+                return;
+            }
+
+            timeline.RelativeChildSize = new Vector2((float)Math.Max(1, track.Length), 1);
+        }
+
         protected void Add(Drawable visualisation) => timeline.Add(visualisation);
 
         protected virtual void LoadBeatmap(WorkingBeatmap beatmap)
         {
+            timeline.Clear();

```

Comment:
Reason for moving critical functionality into a virtual method?

Reply:
see the commit: it's to fix the marker never appearing

Reply:
Wouldn't you say the proper fix is to re-load the marker?

Reply:
recreate a drawable that never changed? I dunno, seems wrong to me

an alternative if you don't want to put this in the virtual method (which I implemented at first) is to make a bool field/property `ShouldClearOnBeatmapChange`.

Reply:
We're moving in that direction in general - every OsuScreen is reloaded when entered, Player is going to be entirely reloaded if mods are changed, etc...

Reply:
where necessary, definitely. I don't see it as being necessary at this level. if it was the whole *timeline itself* being reloaded then sure, but not individual components inside. it means we can't do transitions in cases where they could be wanted

Reply:
consider in this case, we probably want the marker to move with a transition rather than jumping to the new time. this is a pretty common use pattern in all the drawable components we have.

components themselves can be reloaded, but pieces inside them shouldn't unnecessarily be

Reply:
This is going to be the case depending on how the audio is seeked, because the marker is seeked to the audio time.